redis.conf

```diff
@@ -70,6 +70,29 @@ tcp-backlog 511
 # unixsocket /tmp/redis.sock
 # unixsocketperm 700
 
+# Enable forcing of TCP connections over SSL. 
+# If enabled, you must specify either a certificate root store file or directory (not both) in 
+# PEM format (or a set of PEM files if a directory). You must also specify the certificate for the
+# server, the private key file, and the Diffie-Hellman parameters file. These may be in separate files
+# or in the same file. All files must be in PEM format (as required by openSSL). 
+# If you wish, you can customize the cipher list to use when accepting SSL connections. By default, the list
+# By default, we use all ciphers with the exception of: ones with anonymous Diffie-Hellman, low-strength ciphers, 
+# export ciphers, md5 hashing. Ordered from strongest to weakest.
+# Note that as ciphers become broken, it will be necessary to change the available cipher list to remain secure. 
+#
+# If you are using slave processes, and wish to validate the common name of the redis server, you can
+# specify the parameter ssl_cert_common_name with the name you wish to validate against.
+#
+# ssl true
+# ssl_ca_root_dir < Path to your openSSL root certificates directory >
+# ssl_ca_root_file < Path to your openSSL root certificate file (You must specify either a root CA file, or a directory, but not both) >
+# ssl_cert_file <Path to your server's SSL cert>
+# ssl_pk_file <Path to your server's private key file (may be the same as above)>
+# ssl_dhk_file <Path to your Diffie-Hellman data file (may be the same as above)>
+# ssl_cert_pass <optional - password for server cert file>
+# ssl_cipher_list <optional - cipher list to use for the server. Defaults to ALL:!ADH:!LOW:!EXP:!MD5:@STRENGTH>
+# ssl_cert_common_name <optional - common name of server cert to match>
+
```

Comment:
Adding my comment here since this PR was linked from #2178:

This seems a bit confusing to me. Why load all certificates from a directory? At least from a user's perspective, I would recommend taking a look at the MongoDB configuration options 
http://docs.mongodb.org/manual/reference/configuration-options/#net-ssl-options

Being able to set a mode is nice and would allow you to not opt everyone into using SSL. Also there are varying degress of SSL, do you just allow it if available, do you prefer it and only allow SSL, or do you also have strict requirements where everything should be validated against the CA certificate.